internal/core/data/v2/handler/event.go

```diff
@@ -0,0 +1,82 @@
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package handler
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+
+	dataContainer "github.com/edgexfoundry/edgex-go/internal/core/data/container"
+	v2DataContainer "github.com/edgexfoundry/edgex-go/internal/core/data/v2/bootstrap/container"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
+
+	"github.com/edgexfoundry/go-mod-bootstrap/bootstrap/container"
+	"github.com/edgexfoundry/go-mod-bootstrap/di"
+
+	"github.com/edgexfoundry/go-mod-core-contracts/clients"
+	model "github.com/edgexfoundry/go-mod-core-contracts/v2/models"
+
+	msgTypes "github.com/edgexfoundry/go-mod-messaging/pkg/types"
+)
+
+// The AddEvent function accepts the new event model from the controller functions
+// and invokes addEvent function in the infrastructure layer
+func AddEvent(e model.Event, ctx context.Context, dic *di.Container) (string, error) {
+	configuration := dataContainer.ConfigurationFrom(dic.Get)
+	dbClient := v2DataContainer.DBClientFrom(dic.Get)
+
+	err := checkDevice(e.Device, ctx, dic)
+	if err != nil {
+		return "", err
+	}
+
+	// Add the event and readings to the database
+	if configuration.Writable.PersistData {
+		addedEvent, err := dbClient.AddEvent(e)
```

Comment:
@lenny-intel  Hi Lenny, as discussed @cloudxxx8 , `dbClient.AddEvent(e)` now returns the event model instead of simply id in v2 API implementation so that we can retrieve the `id`, `created` field added in the db layer as Mike suggested in https://github.com/edgexfoundry/edgex-go/pull/2605#discussion_r451734825. 
It seems that is making more sense that the `created` time stamp is assigned in the db layer (the time when the entry is created in DB).
Would you please let us know what you think about this? Thanks.